docket_scrap/spiders/docket_spider.py

```diff
@@ -0,0 +1,179 @@
+"""
+Scrapy Spider that crawls RIPUC webpage for commission dockets
+run by command "scrapy crawl docket" on cmd
+"""
+
+from w3lib.html import remove_tags
+import re
+import scrapy
+from urllib.parse import urljoin
+import csv
+
+from ..items import Docket
+
+# from scrapy.http.request import Request
+
+
+class DocketSpider(scrapy.Spider):
+    name = 'docket'
+    start_urls = [
+        'http://www.ripuc.ri.gov/eventsactions/docket.html'
+    ]
+    base_url = 'http://www.ripuc.ri.gov/eventsactions/'
+
+    def parse(self, response):
+        """
+            Handling the response of the webpage
+        """
+        rows = response.css("table")[2].css("table")[1].xpath('tr')
+
+        '''
+            temporary storage for dockets to check if there is any missing field
+        '''
+        docket_temporary_storage = dict()
+
+        for row in rows:
+            col = row.css('td')
+
+            if len(col) == 3:
+                '''
+                    when a row has three columns, it must have an ID, a Filer and a Description
+                '''
+
+                def extract_from_three_columns():
+                    return self.create_docket(
+                        docket_id=self.extract_docket_id(col[0]),
+                        description=self.extract_description(col[2]),
+                        date=self.extract_filing_date(col[2]),
+                        filer=self.extract_filer(col[1]),
+                        file_url=self.extract_links(col[0])
+                    )
+
+                docket = extract_from_three_columns()
+                docket_temporary_storage = docket
+
+            elif len(col) == 2:
+
+                def extract_from_two_columns():
+                    docket_id = self.extract_docket_id(col[0])
+
+                    if docket_id:
+                        ''' if id is present, then the filer is merged with previous id '''
+                        return self.create_docket(
+                            docket_id=docket_id,
+                            description=self.extract_description(col[1]),
+                            date=self.extract_filing_date(col[1]),
+                            filer=docket_temporary_storage["filer"],
+                            file_url=self.extract_links(col[0])
+                        )
+
+                    else:
+                        ''' if id not present, then it is merged with previous id '''
+                        return self.create_docket(
+                            docket_id=docket_temporary_storage["docket_id"],
+                            description=self.extract_description(col[1]),
+                            date=self.extract_filing_date(col[1]),
+                            filer=self.extract_filer(col[0]),
+                            file_url=docket_temporary_storage["file_url"]
```

Comment:
same here.no need to create function. Set docket_ud, filer, and file_url in if/else condition and then call create_docket.